### ISSUE TITLE
Game crashes when joining/loading a chunk with certain block entities

### ISSUE THREAD

**Issue by SamGamerYT** [GeyserMC/Geyser#4240]
### Describe the bug

Player crashes when they go to spawn, doesn't matter where unless it's a vanilla world then they won't get crashed - I'm not personally sure if it's the NPC, items or holograms that's crashing them. wasn't a thing before 1.20.40 update,

### To Reproduce

Go to /spawn
*Gets crashed*

### Expected behaviour

Go to /spawn
*Enjos their time*

### Screenshots / Videos

[17:32:05 INFO] [geyser]: Golden Fox313 has disconnected from the Java server because of Bedrock client disconnected
[17:31:27 INFO] [geyser]: QueenChaos95180 has disconnected from the Java server because of Bedrock client timed out

### Server Version and Plugins

git-Purpur-2062 (MC: 1.20.1)

### Geyser Dump

https://dump.geysermc.org/UgNY4VKAUDZaReRWbP2OEvKdfcaTtDPG

### Geyser Version

Master Branch - 339

### Minecraft: Bedrock Edition Device/Version

Playstation 4, Xbox One X

**Comment by SamGamerYT** [comment]
I don't have ViaVersion on proxy any longer, it's related to scoreboard as I have tried disabling it

**Comment by Kas-tle** [comment]
Could you please test with an older version to confirm this is 1.20.40 related? Also please test on a non-console platform to see if this is universal.

**Comment by SamGamerYT** [comment]
> Could you please test with an older version to confirm this is 1.20.40 related? Also please test on a non-console platform to see if this is universal.

It works as expected on the older version with over 50-60 players playing, now if ANYONE leaves their small island and go to a bigger land on the newer version, crash happens.

**Comment by SamGamerYT** [comment]
> Could you please test with an older version to confirm this is 1.20.40 related? Also please test on a non-console platform to see if this is universal.

no crash happens on Windows 10, Crash does however happens on Mobile, Xbox, PS, I don't have switch friends. lend me some switch friends.

**Comment by Konicai** [comment]
Is this contained within a single backend server or does it involve server switching?

**Comment by onebeastchris** [comment]
Just curious; also when switching dimensions? (overworld/nether/end), or just when switching servers?

**Comment by SamGamerYT** [comment]
> Just curious; also when switching dimensions? (overworld/nether/end), or just when switching servers?

overworld

**Comment by SamGamerYT** [comment]
> No, i mean like, does it also crash when going from overworld -> nether? (aka, when switching dimensions)

bit hard to do that as right now everyone just crashes the moment they join xd

**Comment by SamGamerYT** [comment]
> No, i mean like, does it also crash when going from overworld -> nether? (aka, when switching dimensions)

no crash

**Comment by newbNox** [comment]
> > No, i mean like, does it also crash when going from overworld -> nether? (aka, when switching dimensions)
> 
> bit hard to do that as right now everyone just crashes the moment they join xd

I have the very same issue, of player's game crashing right after they join

**Comment by Noobieswat** [comment]
Same issue but on Server switching but only to one of 3 Servers.

**Comment by Noobieswat** [comment]
Only difference I know on my setup is Multiverse + Plotsqared on the one Server with the Problem.

**Comment by mastachef** [comment]
Happening to me as well

**Comment by Noobieswat** [comment]
on Android on Version 1.20.32 all fine. Same device Just Updated to Bedrock error Happens after Update to 1.20.40 Game just crashes.
And sorry for my grammar

**Comment by onebeastchris** [comment]
We are aware of the issue and looking into a fix. Please don't bump this issue unless you have further details that might help narrow the bug down.

**Comment by SamGamerYT** [comment]
[18:54:51 INFO] [geyser]: BrokenJJ6029 has disconnected from the Java server because of Bedrock client timed out
[18:54:51 INFO]: [connected player] _BrokenJJ6029 (/Pervert Looking at IP?) has disconnected: An internal error occurred in your connection.
[18:54:51 ERROR]: [connected player] _BrokenJJ6029 (/Pervert Looking at IP?): exception encountered in com.velocitypowered.proxy.connection.client.ClientPlaySessionHandler@1e310c56
java.nio.channels.ClosedChannelException: null
        at io.netty.channel.local.LocalChannel.doWrite(LocalChannel.java:368) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:931) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:895) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1372) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:921) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:907) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:893) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.DefaultChannelPipeline.flush(DefaultChannelPipeline.java:967) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannel.flush(AbstractChannel.java:254) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at org.geysermc.geyser.network.netty.ChannelWrapper.flush(ChannelWrapper.java:213) ~[?:?]
        at com.velocitypowered.proxy.connection.MinecraftConnection.flush(MinecraftConnection.java:244) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at com.velocitypowered.proxy.connection.backend.BackendPlaySessionHandler.readCompleted(BackendPlaySessionHandler.java:324) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at com.velocitypowered.proxy.connection.MinecraftConnection.channelReadComplete(MinecraftConnection.java:161) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:486) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:463) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelReadComplete(AbstractChannelHandlerContext.java:456) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at com.velocitypowered.proxy.protocol.netty.AutoReadHolderHandler.channelReadComplete(AutoReadHolderHandler.java:69) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:484) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:463) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelReadComplete(AbstractChannelHandlerContext.java:456) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.handler.timeout.IdleStateHandler.channelReadComplete(IdleStateHandler.java:295) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:484) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:463) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelReadComplete(AbstractChannelHandlerContext.java:456) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.handler.codec.ByteToMessageDecoder.channelReadComplete(ByteToMessageDecoder.java:358) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:486) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:463) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelReadComplete(AbstractChannelHandlerContext.java:456) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelReadComplete(DefaultChannelPipeline.java:1415) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:482) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelReadComplete(AbstractChannelHandlerContext.java:463) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.DefaultChannelPipeline.fireChannelReadComplete(DefaultChannelPipeline.java:925) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:820) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[velocity.jar:3.2.0-SNAPSHOT (git-19abb909-b265)]
        at java.lang.Thread.run(Thread.java:833) [?:?]
[18:54:51 INFO] [floodgate]: Floodgate player logged in as BrokenJJ6029 disconnected

**Comment by SamGamerYT** [comment]
One of the users dc'ed and that's a first time I saw an error related to the timed out

**Comment by Noobieswat** [comment]
Im still trying to find it but it seems on my server its about Chisseld Bookshelves but after I deleted every one Players again could join

**Comment by Noobieswat** [comment]
When they are out of veiw (Chisseld Bookshelves) the game Crashes on Join

**Comment by SamGamerYT** [comment]
> When they are out of veiw (Chisseld Bookshelves) the game Crashes on Join

hmmm not the case for me, i got 0 any type of bookshelves

**Comment by Noobieswat** [comment]
Maybee a block data type something like that

**Comment by SamGamerYT** [comment]
Hi hi, it was red_bed, bookshelf and chiseled_bookshelf - that caused the crashes apparently?

**Comment by LemonjamesD** [comment]
I have no red beds, bookshelves, nor chiseledbookshelves and my bedrock friend is still crashing.

**Comment by thiagogebrimm** [comment]
same problem

**Comment by diamond-gold** [comment]
Client crash due to missing block entity

**Comment by cmrmarcelo** [comment]
Bom fiz alguns testes ,espero que ajude

**Comment by LenyzDev** [comment]
@cmrmarcelo use english

**Comment by VariedGamerYT** [comment]
So ngl I’m a bedrock player and got no clue what any of this means, but ngl this is affecting almost half of the 100 players server I’m in and is there any possible way of a fix in sight yet? (Not sound impatient)

**Comment by cmrmarcelo** [comment]
Well, I did some tests, I hope it helps. From what I could observe before falling, the chunk I teleported to had ejector funnels turned off, observer pistons and chests and spawners

**Comment by Camotoy** [comment]
We pushed an update several hours ago that may have fixed your issue. Please try downloading the new build.

**Comment by VariedGamerYT** [comment]
Well I hope they find a way to fix this soon bc the server is slowly
starting to be removed and we have no fix.

**Comment by HalfloveJAR** [comment]
After the latest Geyser update I can confirm that I am no longer having this issue on my 1.20.1 Purpur #2062 server 👍

**Comment by nakanotti** [comment]
I have confirmed that the problem no longer occurs after [8d2ebcf](https://github.com/GeyserMC/Geyser/commit/8d2ebcf69bee0b329c6c748845113836f160319e) .

**Comment by yichifauzi** [comment]
the problem still occurs when using velocity proxy...... and my backend server using paper 1.20.1 git-Paper-773dd72....
already using latest velocity

**Comment by TuxGamer** [comment]
The problem is still occurring with Paper 1.20.2, latest build

`git-Paper-249 (MC: 1.20.2) (Implementing API version 1.20.2-R0.1-SNAPSHOT) (Git: 996d529)`

**Comment by JHarris12345** [comment]
I am getting the problem on 1.20.2 bungeecord. I have noticed that removing floodgate from my hub server does NOT crash me anymore. Instead it disconnects me with a QuietException and the console prints:

[23:07:07 INFO]: ^Floodgate^>Tp8K9WdTvpyjIMyD!ZbojkbU79a5g3Cpe3qwYnoJZ5oc1w7Ok/jizISUT5rP3FoIWeYmxDCGrsaVGX6wqeD1n4m3i+/tUL6r6zontQqjyQJP4/5AF6Drie4g+mTM6wNbnval21u4eIjLLIAIp6Xnlxy75:34798 lost connection: Internal Exception: io.netty.handler.codec.DecoderException: java.io.IOException: Packet 0/0 (PacketPlayInTeleportAccept) was larger than I expected, found 12 bytes extra whilst reading packet 0

When putting floodgate back on, it's back to crashing

**Comment by MaPa80711** [comment]
Hello, I can also confirm that after the adjustments in https://github.com/GeyserMC/Geyser/commit/8d2ebcf69bee0b329c6c748845113836f160319e
the client application no longer crashes.

This applies to PS4/PS5 and Switch Minecraft version 1.20.40.

But there is a temporary problem with the consoles mentioned as soon as you open the chat and want to enter a command starting with "/". As soon as you type "/" on the console, the client crashes again.

**Comment by onebeastchris** [comment]
Unfortunately, the slash command issue is known and unrelated.
See https://github.com/GeyserMC/Geyser/issues/3990 for solutions

**Comment by VeerDhindsa** [comment]
im running a geyser server and whenever i load a chunk at my base, the game just crashes, im using fabric 1.20.2 and the latest geysermc

**Comment by GabrielMoreiradeSouza** [comment]
Players on my server have been experiencing an issue for some time. Bedrock players are playing normally, but suddenly they get disconnected and are sent back to the hub.

**Comment by Kas-tle** [comment]
> Players on my server have been experiencing an issue for some time. Bedrock players are playing normally, but suddenly they get disconnected and are sent back to the hub.

If players are not having their client crash this is unrelated.

**Comment by CheeseManFuu** [comment]
Just commented in the ViaVersion thinking the addition of their plugin was what was killing Bedrock players for some reason.

Build 340 which I didn't notice until just maybe half an hour ago has fixed crashing connections on Pocket Edition (Android), Switch, and Windows 10.

**Comment by Tgwizman** [comment]
I run a server with some friends. Tracked down that it was always kicking bedrock players when near a specific location. One spot was at spawn, another was a few km west of spawn (people build there). I copied all the server files over to another server and got the same issue until I removed the map. I then transferred all the files except the region files, no issue. I then tracked it down to `r.0.0.mca`. Someone teleported a bedrock player to spawn, the issue was gone with the region. Then they tried teleporting them to the base 3km away and it crashed again.

TL;DR
It's definitely effecting bedrock players when they load in chunk data on the client-side. Whatever _it_ is...

**Comment by VeerDhindsa** [comment]
The Problem Is Gone After I Installed Build 340 On My Server, Still Wondering What Caused This IssueSent from my iPhoneOn Oct 29, 2023, at 23:58, CheeseManFuu ***@***.***> wrote:﻿
Just commented in the ViaVersion thinking the addition of their plugin was what was killing Bedrock players for some reason.
Build 340 which I didn't notice until just maybe half an hour ago has fixed crashing connections on Pocket Edition (Android), Switch, and Windows 10.

**Comment by CheeseManFuu** [comment]
> The Problem Is Gone After I Installed Build 340 On My Server, Still Wondering What Caused This Issue

probably just typical Java-Bedrock interplay causing issue. It didn't happen until I updated the server plugins to include Build 339 which opened up Bedrock 1.20.40 support, so probably something along the lines of the new Bedrock version not playing nice.

**Comment by Kas-tle** [comment]
This issues has been resolved by Bedrock Client version 1.20.41. Please advise any play set still experiencing this issue to upgrade their client.

**Comment by Agaloth** [comment]
This issue is not fixed @Kas-tle (in this video I teleported to a player warp before hand and my game just got locked, I crash every time I join now (resource pack or not)https://github.com/GeyserMC/Geyser/assets/33335971/3334f7a8-2682-42af-b365-9916ed05e384

**Comment by Agaloth** [comment]
I'm also using 1.20.41 and the updated geyser standalone today.
